stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt

```diff
@@ -101,39 +104,36 @@ public fun Messages(
         }
         val focusedItemIndex = messages.indexOfFirst { it is MessageItemState && it.focusState is MessageFocused }
 
-        if (focusedItemIndex != -1) {
-            coroutineScope.launch {
+        LaunchedEffect(key1 = messagesState) {
+            if (focusedItemIndex != -1) {
                 currentListState.scrollToItem(focusedItemIndex)
             }
-        }
 
-        when {
-            !currentListState.isScrollInProgress && newMessageState == Other && firstVisibleItemIndex < 3 -> {
-                coroutineScope.launch {
```

Comment:
This refactor is a consequence of mitigating the warning "Calls to launch should happen inside a LaunchedEffect and not composition" that is now displayed when running Compose 1.1.0-rc03.

The official take is that `rememberCoroutineScope` should be used only if you are launching a Coroutine outside of Composition but want that Coroutine to share the same Lifecycle as the Composable.

The refactor above function mostly the same but it has a notable flaw that we have encountered previously which is failing to kep up with the influx of new messages in a fast live stream enviroment.